moveit_core/planning_scene/src/planning_scene.cpp

```diff
@@ -448,8 +448,12 @@ void PlanningScene::pushDiffs(const PlanningScenePtr& scene)
       if (it.second == collision_detection::World::DESTROY)
       {
         scene->world_->removeObject(it.first);
-        scene->removeObjectColor(it.first);
-        scene->removeObjectType(it.first);
+        if (!robot_state_->hasAttachedBody(it.first))
```

Comment:
Yep, we use `DESTROY`.  But looks like the other change is enough to restore the color of detached objects.
I can undo this one, but not sure what will be the implications

Reply:
The other change doesn't "restore" color, it just doesn't delete it from the world upon detaching. The color was left in the world during attaching.

I would prefer if the color was removed from the world during attaching, and added to the world using the info in the `AttachedBody` during detaching. But if you don't have the cycles to add that, this will work, too.

Reply:
I'd just add a comment to the effect of `// Workaround: Remember color in world while object is attached to robot, instead of recovering the info from AttachedBody`